kit/ChildSession.cpp

```diff
@@ -2464,12 +2466,31 @@ bool ChildSession::renderNextSlideLayer(SlideCompressor &scomp,
             output.reserve(response.size() + pixmap->size());
             output.resize(response.size());
             std::memcpy(output.data(), response.data(), response.size());
+            std::vector<char> compressedOutPut;
+            compressedOutPut.reserve(pixmap->size());
+            compressedOutPut.resize(pixmap->size());
```

Comment:
If you are using resize and reserve with the same size then there isn't really any point to reserve. I'd drop that line and just use resize (and then use compressedOutPut.size() as arg 2 to ZSTD_compress)

But I think you should resize to ZSTD_COMPRESSBOUND(pixmap->size()), as the upper bound of the size needed to compress pixmap->size().

I think we need to add extra arguments to the json message to send the width and height of the image because this isn't a png anymore where we could get that info from the png itself.